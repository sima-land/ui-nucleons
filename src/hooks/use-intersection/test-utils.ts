import { act } from '@testing-library/react';

/**
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export type IntersectionObserverEntryBase = Omit<
  IntersectionObserverEntry,
  'time' | 'rootBounds' | 'boundingClientRect' | 'intersectionRect'
>;

let intersectionMock: IntersectionMock | undefined;

/**
 * Возвращает новый объект, частично реализующий IntersectionObserverEntry.
 * @param base Базовая информация.
 * @return Объект.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export function createFakeEntry(base: IntersectionObserverEntryBase): IntersectionObserverEntry {
  const entry: IntersectionObserverEntry = {
    ...base,

    time: Date.now(),

    /** @inheritdoc */
    get rootBounds() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },

    /** @inheritdoc */
    get boundingClientRect() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },

    /** @inheritdoc */
    get intersectionRect() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },
  } as unknown as IntersectionObserverEntry;

  return entry;
}

/**
 * Реализует интерфейс для управления состоянием пересечения элемента с область видимости.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export class IntersectionMock {
  _original: typeof IntersectionObserver;
  _observers: FakeIntersectionObserver[] = [];

  /** Конструктор. */
  constructor() {
    this._original = window.IntersectionObserver;
  }

  /**
   * Подменяет глобальный конструктор IntersectionObserver.
   */
  apply() {
    if (intersectionMock) {
      throw Error('Already mocked');
    }

    // eslint-disable-next-line @typescript-eslint/no-this-alias
    intersectionMock = this;
    window.IntersectionObserver = FakeIntersectionObserver;
  }

  /**
   * Восстанавливает оригинальный глобальный конструктор IntersectionObserver.
   */
  restore() {
    if (window.IntersectionObserver !== FakeIntersectionObserver) {
      throw Error('Not mocked');
    }

    window.IntersectionObserver = this._original;
    intersectionMock = undefined;
  }

  /**
   * Имитирует изменение состояния пересечения элемента с областью видимости.
   * @param entryArg Данные пересечения элемента.
   */
  changeElementState(entryArg: IntersectionObserverEntryBase) {
    act(() => {
      const entry = createFakeEntry(entryArg);

      for (const observer of this._observers) {
        if (
          observer._elements.some(e => e === entry.target) &&
          observer.thresholds.includes(entry.intersectionRatio)
        ) {
          (observer as FakeIntersectionObserver)._runCallback(entry);
        }
      }
    });
  }

  /**
   * Записывает observer в список зарегистрированных.
   * @param observer Observer.
   */
  _registerObserver(observer: FakeIntersectionObserver) {
    this._observers.push(observer);
  }

  /**
   * Удаляет observer из списка зарегистрированных.
   * @param observer Observer.
   */
  _unregisterObserver(observer: FakeIntersectionObserver) {
    this._observers.splice(this._observers.indexOf(observer), 1);
  }
}

/**
 * Имитация IntersectionObserver.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export class FakeIntersectionObserver implements IntersectionObserver {
  readonly root: Element | Document | null;
  readonly rootMargin: string;
  readonly thresholds: ReadonlyArray<number>;

  _callback: IntersectionObserverCallback;
  _elements: Element[];

  /** @inheritdoc */
  constructor(callback: IntersectionObserverCallback, options: IntersectionObserverInit = {}) {
    this.root = options?.root || null;
    this.rootMargin = options?.rootMargin || '0px 0px 0px 0px';
    this.thresholds = Array.isArray(options.threshold)
      ? options.threshold
      : [options.threshold || 0];

    this._callback = callback;
    this._elements = [];
    intersectionMock?._registerObserver(this);
  }

  /** @inheritdoc */
  observe(element: Element) {
    this._elements.push(element);
  }

  /** @inheritdoc */
  unobserve(element: Element) {
    this._elements = this._elements.filter(e => e !== element);
  }

  /** @inheritdoc */
  disconnect() {
    this._elements = [];
    intersectionMock?._unregisterObserver(this);
  }

  /** @inheritdoc */
  takeRecords() {
    // eslint-disable-next-line no-console
    console.error('Not implemented in mock');

    return [];
  }

  /**
   * Запускает callback.
   * @param entry Данные пересечения элемента.
   */
  _runCallback(entry: IntersectionObserverEntry) {
    this._callback([entry], this);
  }
}
