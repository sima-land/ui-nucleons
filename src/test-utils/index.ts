export { IntersectionObserverMock } from './intersection-observer-mock';
export { ResizeObserverMock } from './resize-observer-mock';
export { createDOMRectReadOnly } from './create-dom-rect-read-only';

// @todo matchMedia mock utils
