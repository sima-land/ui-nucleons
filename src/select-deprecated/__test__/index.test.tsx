import { mount, ReactWrapper } from 'enzyme';
import { act, Simulate } from 'react-dom/test-utils';
import { Select } from '..';
import { TextField } from '../../text-field';
import { render, fireEvent } from '@testing-library/react';
import GoogleSVG from '@sima-land/ui-quarks/icons/24x24/Filled/Social/Google';

describe('Select', () => {
  const openMenu = (wrapper: ReactWrapper) => {
    act(() => {
      Simulate.keyDown(
        wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
        { key: 'Enter' },
      );
    });
    wrapper.update();
  };

  const findMenu = (wrapper: ReactWrapper) => wrapper.find('div[data-testid="select:menu"]');

  it('should renders correctly', () => {
    const menuToggleSpy = jest.fn();
    const selectSpy = jest.fn();
    const clickSpy = jest.fn();

    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        onMenuToggle={menuToggleSpy}
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
        optionSize='m'
        onSelect={selectSpy}
        onClick={clickSpy}
        renderOption={option => option.toUpperCase()}
      />,
    );

    expect(wrapper.getDOMNode()).toMatchSnapshot();
    openMenu(wrapper);
    expect(wrapper.getDOMNode()).toMatchSnapshot();
  });

  it('should handle TextField block click', () => {
    const clickSpy = jest.fn();

    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
        onClick={clickSpy}
      />,
    );

    expect(clickSpy).toHaveBeenCalledTimes(0);

    act(() => {
      Simulate.click(wrapper.find(TextField).find('[data-testid="text-field:block"]').getDOMNode());
    });
    wrapper.update();

    expect(clickSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle TextField mousedown', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
      />,
    );

    act(() => {
      Simulate.mouseDown(
        wrapper.find(TextField).find('[data-testid="text-field:block"]').getDOMNode(),
      );
    });
    wrapper.update();

    expect(wrapper.find('div[data-testid="select:menu"]')).toHaveLength(1);

    act(() => {
      Simulate.mouseDown(
        wrapper.find(TextField).find('[data-testid="text-field:block"]').getDOMNode(),
      );
    });
    wrapper.update();

    expect(wrapper.find('div[data-testid="select:menu"]')).toHaveLength(0);
  });

  it('should handle TextField enter key down', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
      />,
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);
  });

  it('should render loading state', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
        loading
      />,
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);

    expect(wrapper.getDOMNode()).toMatchSnapshot();
  });

  it('should handle menu enter key down', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
      />,
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);

    act(() => {
      Simulate.keyDown(findMenu(wrapper).getDOMNode(), { key: 'Enter' });
    });
    wrapper.update();
    expect(findMenu(wrapper)).toHaveLength(0);
  });

  it('should handle menu not enter key down', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
      />,
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);

    act(() => {
      Simulate.keyDown(findMenu(wrapper).getDOMNode(), { key: 'E' });
    });
    wrapper.update();
    expect(findMenu(wrapper)).toHaveLength(1);
  });

  it('should handle menu blur', () => {
    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
      />,
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);

    act(() => {
      Simulate.blur(findMenu(wrapper).getDOMNode());
    });
    wrapper.update();
    expect(findMenu(wrapper)).toHaveLength(0);
  });

  it('should handle menu item select', () => {
    const selectSpy = jest.fn();

    const wrapper = mount(
      <Select
        label='Формат каталога'
        value='Карточки товаров JPG'
        options={[
          'Прайс-лист XML',
          'Прайс-лист YML',
          'Карточки товаров JPG',
          'Фото товаров JPG + прайс-лист XLS',
          'Печатный каталог PDF',
        ]}
        optionSize='m'
        onSelect={selectSpy}
      />,
    );

    expect(selectSpy).toHaveBeenCalledTimes(0);

    openMenu(wrapper);

    expect(selectSpy).toHaveBeenCalledTimes(0);

    act(() => {
      Simulate.click(wrapper.find('div[role="menuitem"]').at(2).getDOMNode());
    });
    wrapper.update();

    expect(selectSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle ReactNode in "endAdornment" prop', () => {
    const { getAllByTestId } = render(
      <Select
        label='Формат каталога'
        value='AAAA'
        options={['AAAA', 'BBBB', 'CCCC', 'DDDD', 'EEEE']}
        endAdornment={<GoogleSVG data-testid='google-svg' />}
      />,
    );

    expect(getAllByTestId('google-svg')).toHaveLength(1);
  });

  it('should handle function in "endAdornment" prop', () => {
    const { getAllByTestId, getByTestId } = render(
      <Select
        label='Формат каталога'
        value='AAAA'
        options={['AAAA', 'BBBB', 'CCCC', 'DDDD', 'EEEE']}
        endAdornment={opened => (
          <GoogleSVG data-testid={`google-svg-${opened ? 'opened' : 'closed'}`} />
        )}
      />,
    );

    expect(getAllByTestId('google-svg-closed')).toHaveLength(1);

    fireEvent.mouseDown(getByTestId('text-field:block'));

    expect(getAllByTestId('google-svg-opened')).toHaveLength(1);
  });
});
