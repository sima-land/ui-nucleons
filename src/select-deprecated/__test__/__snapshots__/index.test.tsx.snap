// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render loading state 1`] = `
<Select
  label="Формат каталога"
  loading={true}
  options={
    [
      "Прайс-лист XML",
      "Прайс-лист YML",
      "Карточки товаров JPG",
      "Фото товаров JPG + прайс-лист XLS",
      "Печатный каталог PDF",
    ]
  }
  value="Карточки товаров JPG"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      baseInputProps={
        {
          "style": {
            "pointerEvents": "none",
          },
          "tabIndex": -1,
        }
      }
      blockProps={
        {
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "ref": {
            "current": <div
              class="reset input-block focused variant-desktop size-l with-label rounds-sm__all"
              data-testid="text-field:block"
              tabindex="0"
            >
              <div
                class="main-column"
              >
                <label
                  class="label focused variant-desktop filled"
                  data-testid="text-field:label"
                >
                  Формат каталога
                </label>
                <div
                  class="input-wrapper"
                >
                  <div
                    class="reset root base-input focused variant-desktop size-l"
                    data-testid="base-input:root"
                    style="pointer-events: none;"
                  >
                    <input
                      class="field reset"
                      data-testid="text-field:field"
                      readonly=""
                      tabindex="-1"
                      value="Карточки товаров JPG"
                    />
                  </div>
                </div>
              </div>
              <div
                class="display-flex align-items-center M-l__4"
              >
                <svg
                  fill="#9e9e9e"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>,
          },
          "tabIndex": 0,
        }
      }
      className="field"
      data-testid="select:field"
      endAdornment={
        <ForwardRef(UpSVG)
          fill="#9e9e9e"
        />
      }
      focused={true}
      label="Формат каталога"
      multiline={false}
      onClick={[Function]}
      readOnly={true}
      value="Карточки товаров JPG"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="select:field"
      >
        <div
          className="reset input-block focused variant-desktop size-l with-label rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
        >
          <div
            className="main-column"
          >
            <label
              className="label focused variant-desktop filled"
              data-testid="text-field:label"
            >
              Формат каталога
            </label>
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input focused variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                readOnly={true}
                style={
                  {
                    "pointerEvents": "none",
                  }
                }
                tabIndex={-1}
                value="Карточки товаров JPG"
              >
                <div
                  className="reset root base-input focused variant-desktop size-l"
                  data-testid="base-input:root"
                  style={
                    {
                      "pointerEvents": "none",
                    }
                  }
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    readOnly={true}
                    tabIndex={-1}
                    value="Карточки товаров JPG"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <ForwardRef(UpSVG)
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
                    fillRule="evenodd"
                  />
                </svg>
              </ForwardRef(UpSVG)>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
    <ForwardRef(Dropdown)
      className="menu"
      data-testid="select:menu"
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
      style={
        {
          "top": "56px",
        }
      }
      tabIndex={-1}
    >
      <div
        className="root menu shadow-z4 rounds-md__all"
        data-testid="select:menu"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
        style={
          {
            "top": "56px",
          }
        }
        tabIndex={-1}
      >
        <CustomScrollbar
          className="inner"
          overflow={
            {
              "x": "h",
              "y": "s",
            }
          }
        >
          <OverlayScrollbarsComponent
            className="custom-scrollbar inner"
            options={
              {
                "callbacks": {
                  "onScrollStop": [Function],
                },
                "overflowBehavior": {
                  "x": "h",
                  "y": "s",
                },
              }
            }
          >
            <div
              className="os-host"
            >
              <div
                className="os-resize-observer-host"
              />
              <div
                className="os-padding"
              >
                <div
                  className="os-viewport"
                >
                  <div
                    className="os-content"
                  >
                    <DropdownLoading
                      data-testid="select:loading-area"
                    >
                      <div
                        className="root"
                        data-testid="select:loading-area"
                      >
                        <SpinnerSVG
                          size="s"
                        >
                          <svg
                            className="spinner size-s"
                            data-testid="spinner"
                            height={20}
                            stroke="#1f84db"
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <circle
                              className="circle"
                              cx={10}
                              cy={10}
                              r={9}
                            />
                          </svg>
                        </SpinnerSVG>
                      </div>
                    </DropdownLoading>
                  </div>
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-horizontal "
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-vertical"
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar-corner"
              />
            </div>
          </OverlayScrollbarsComponent>
        </CustomScrollbar>
      </div>
    </ForwardRef(Dropdown)>
  </div>
</Select>
`;

exports[`Select should renders correctly 1`] = `
<Select
  label="Формат каталога"
  onClick={[MockFunction]}
  onMenuToggle={
    [MockFunction] {
      "calls": [
        [
          false,
        ],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSelect={[MockFunction]}
  optionSize="m"
  options={
    [
      "Прайс-лист XML",
      "Прайс-лист YML",
      "Карточки товаров JPG",
      "Фото товаров JPG + прайс-лист XLS",
      "Печатный каталог PDF",
    ]
  }
  renderOption={[Function]}
  value="Карточки товаров JPG"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      baseInputProps={
        {
          "style": {
            "pointerEvents": "none",
          },
          "tabIndex": -1,
        }
      }
      blockProps={
        {
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "ref": {
            "current": <div
              class="reset input-block variant-desktop size-l with-label rounds-sm__all"
              data-testid="text-field:block"
              tabindex="0"
            >
              <div
                class="main-column"
              >
                <label
                  class="label variant-desktop filled"
                  data-testid="text-field:label"
                >
                  Формат каталога
                </label>
                <div
                  class="input-wrapper"
                >
                  <div
                    class="reset root base-input variant-desktop size-l"
                    data-testid="base-input:root"
                    style="pointer-events: none;"
                  >
                    <input
                      class="field reset"
                      data-testid="text-field:field"
                      readonly=""
                      tabindex="-1"
                      value="Карточки товаров JPG"
                    />
                  </div>
                </div>
              </div>
              <div
                class="display-flex align-items-center M-l__4"
              >
                <svg
                  fill="#9e9e9e"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>,
          },
          "tabIndex": 0,
        }
      }
      className="field"
      data-testid="select:field"
      endAdornment={
        <ForwardRef(DownSVG)
          fill="#9e9e9e"
        />
      }
      focused={false}
      label="Формат каталога"
      multiline={false}
      onClick={[Function]}
      readOnly={true}
      value="Карточки товаров JPG"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="select:field"
      >
        <div
          className="reset input-block variant-desktop size-l with-label rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
        >
          <div
            className="main-column"
          >
            <label
              className="label variant-desktop filled"
              data-testid="text-field:label"
            >
              Формат каталога
            </label>
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                readOnly={true}
                style={
                  {
                    "pointerEvents": "none",
                  }
                }
                tabIndex={-1}
                value="Карточки товаров JPG"
              >
                <div
                  className="reset root base-input variant-desktop size-l"
                  data-testid="base-input:root"
                  style={
                    {
                      "pointerEvents": "none",
                    }
                  }
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    readOnly={true}
                    tabIndex={-1}
                    value="Карточки товаров JPG"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <ForwardRef(DownSVG)
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                    fillRule="evenodd"
                  />
                </svg>
              </ForwardRef(DownSVG)>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
  </div>
</Select>
`;

exports[`Select should renders correctly 2`] = `
<Select
  label="Формат каталога"
  onClick={[MockFunction]}
  onMenuToggle={
    [MockFunction] {
      "calls": [
        [
          false,
        ],
        [
          true,
        ],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSelect={[MockFunction]}
  optionSize="m"
  options={
    [
      "Прайс-лист XML",
      "Прайс-лист YML",
      "Карточки товаров JPG",
      "Фото товаров JPG + прайс-лист XLS",
      "Печатный каталог PDF",
    ]
  }
  renderOption={[Function]}
  value="Карточки товаров JPG"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      baseInputProps={
        {
          "style": {
            "pointerEvents": "none",
          },
          "tabIndex": -1,
        }
      }
      blockProps={
        {
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "ref": {
            "current": <div
              class="reset input-block focused variant-desktop size-l with-label rounds-sm__all"
              data-testid="text-field:block"
              tabindex="0"
            >
              <div
                class="main-column"
              >
                <label
                  class="label focused variant-desktop filled"
                  data-testid="text-field:label"
                >
                  Формат каталога
                </label>
                <div
                  class="input-wrapper"
                >
                  <div
                    class="reset root base-input focused variant-desktop size-l"
                    data-testid="base-input:root"
                    style="pointer-events: none;"
                  >
                    <input
                      class="field reset"
                      data-testid="text-field:field"
                      readonly=""
                      tabindex="-1"
                      value="Карточки товаров JPG"
                    />
                  </div>
                </div>
              </div>
              <div
                class="display-flex align-items-center M-l__4"
              >
                <svg
                  fill="#9e9e9e"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>,
          },
          "tabIndex": 0,
        }
      }
      className="field"
      data-testid="select:field"
      endAdornment={
        <ForwardRef(UpSVG)
          fill="#9e9e9e"
        />
      }
      focused={true}
      label="Формат каталога"
      multiline={false}
      onClick={[Function]}
      readOnly={true}
      value="Карточки товаров JPG"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="select:field"
      >
        <div
          className="reset input-block focused variant-desktop size-l with-label rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
        >
          <div
            className="main-column"
          >
            <label
              className="label focused variant-desktop filled"
              data-testid="text-field:label"
            >
              Формат каталога
            </label>
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input focused variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                readOnly={true}
                style={
                  {
                    "pointerEvents": "none",
                  }
                }
                tabIndex={-1}
                value="Карточки товаров JPG"
              >
                <div
                  className="reset root base-input focused variant-desktop size-l"
                  data-testid="base-input:root"
                  style={
                    {
                      "pointerEvents": "none",
                    }
                  }
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    readOnly={true}
                    tabIndex={-1}
                    value="Карточки товаров JPG"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <ForwardRef(UpSVG)
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
                    fillRule="evenodd"
                  />
                </svg>
              </ForwardRef(UpSVG)>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
    <ForwardRef(Dropdown)
      className="menu"
      data-testid="select:menu"
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
      style={
        {
          "top": "56px",
        }
      }
      tabIndex={-1}
    >
      <div
        className="root menu shadow-z4 rounds-md__all"
        data-testid="select:menu"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
        style={
          {
            "top": "56px",
          }
        }
        tabIndex={-1}
      >
        <CustomScrollbar
          className="inner"
          overflow={
            {
              "x": "h",
              "y": "s",
            }
          }
        >
          <OverlayScrollbarsComponent
            className="custom-scrollbar inner"
            options={
              {
                "callbacks": {
                  "onScrollStop": [Function],
                },
                "overflowBehavior": {
                  "x": "h",
                  "y": "s",
                },
              }
            }
          >
            <div
              className="os-host"
            >
              <div
                className="os-resize-observer-host"
              />
              <div
                className="os-padding"
              >
                <div
                  className="os-viewport"
                >
                  <div
                    className="os-content"
                  >
                    <DropdownItem
                      key="0"
                      onClick={[Function]}
                      role="menuitem"
                      size="m"
                    >
                      <div
                        className="root size-m"
                        data-testid="dropdown-item"
                        onClick={[Function]}
                        role="menuitem"
                      >
                        <div
                          className="col col-center"
                        >
                          <div
                            className="row-main"
                          >
                            ПРАЙС-ЛИСТ XML
                          </div>
                        </div>
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      key="1"
                      onClick={[Function]}
                      role="menuitem"
                      size="m"
                    >
                      <div
                        className="root size-m"
                        data-testid="dropdown-item"
                        onClick={[Function]}
                        role="menuitem"
                      >
                        <div
                          className="col col-center"
                        >
                          <div
                            className="row-main"
                          >
                            ПРАЙС-ЛИСТ YML
                          </div>
                        </div>
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      key="2"
                      onClick={[Function]}
                      role="menuitem"
                      size="m"
                    >
                      <div
                        className="root size-m"
                        data-testid="dropdown-item"
                        onClick={[Function]}
                        role="menuitem"
                      >
                        <div
                          className="col col-center"
                        >
                          <div
                            className="row-main"
                          >
                            КАРТОЧКИ ТОВАРОВ JPG
                          </div>
                        </div>
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      key="3"
                      onClick={[Function]}
                      role="menuitem"
                      size="m"
                    >
                      <div
                        className="root size-m"
                        data-testid="dropdown-item"
                        onClick={[Function]}
                        role="menuitem"
                      >
                        <div
                          className="col col-center"
                        >
                          <div
                            className="row-main"
                          >
                            ФОТО ТОВАРОВ JPG + ПРАЙС-ЛИСТ XLS
                          </div>
                        </div>
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      key="4"
                      onClick={[Function]}
                      role="menuitem"
                      size="m"
                    >
                      <div
                        className="root size-m"
                        data-testid="dropdown-item"
                        onClick={[Function]}
                        role="menuitem"
                      >
                        <div
                          className="col col-center"
                        >
                          <div
                            className="row-main"
                          >
                            ПЕЧАТНЫЙ КАТАЛОГ PDF
                          </div>
                        </div>
                      </div>
                    </DropdownItem>
                  </div>
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-horizontal "
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-vertical"
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar-corner"
              />
            </div>
          </OverlayScrollbarsComponent>
        </CustomScrollbar>
      </div>
    </ForwardRef(Dropdown)>
  </div>
</Select>
`;
